reactor-core/src/main/java/reactor/core/publisher/Operators.java

```diff
@@ -2574,6 +2574,205 @@ private LiftFunction(@Nullable Predicate<Publisher> filter,
 		}
 	}
 
+	/*package*/ static class MonoInnerProducerBase<O>
+			implements InnerProducer<O> {
+
+		private final CoreSubscriber<? super O> actual;
+
+		/**
+		 * The value stored by this Mono operator.
+		 */
+		@Nullable
+		private volatile O   value;
```

Comment:
value can be non-volatile

Reply:
Can it really ? Does a CAS on `state` create a barrier that affects every field?

Reply:
Yeah. It does, value is written before write barriers. Also in java (compares to cpp) when we do compareAndSet we use [`memory_order_acq_rel`](https://stackoverflow.com/questions/7363462/what-are-the-similarities-between-the-java-memory-model-and-the-c11-memory-mod) type of barrier which guarantees that all writes will not be reordered with that barrier and all subsequent read happened with `memory_order_acquire` will see nonvolatile changes happened before ``memory_order_acq_rel` barrier (see for more info -> https://en.cppreference.com/w/cpp/atomic/memory_order)